src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -159,7 +159,13 @@ public class RemoteExecutionService {
   private final Reporter reporter;
   private final boolean verboseFailures;
   private final Path execRoot;
-  private final RemotePathResolver remotePathResolver;
+
+  /**
+   * Do not use directly, instead use the per-spawn resolver created in {@link
+   * #buildRemoteAction(Spawn, SpawnExecutionContext)}.
+   */
+  private final RemotePathResolver baseRemotePathResolver;
```

Comment:
Educational question since I'm not super-familiar with the executor interfaces.

`RemotePathResolver` handles all inputs and outputs? Including virtual inputs like `ParamFileActionInput`? I believe the latter needed special processing for Google's executor so just want to sanity check. It'd be great if this model is simpler.

Reply:
@tjgq is in a better position to answer these questions, but as per my understanding this is the case: `RemotePathResolver` is used for everything here, including parameter files.